EIPS/eip-7509.md

```diff
@@ -0,0 +1,309 @@
+     */
+    function getSystemState(address _systemAddress)
+        external
+        view
+        returns (bool);
+}
+```
+
+**IComponent.sol**
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity >=0.8.0;
+import "./Types.sol";
+
+interface IComponent {
+    /**
+     * The world contract address registered by the component.
+     * @return world contract address.
+     */
+    function world() external view returns (address);
+
+    /**
+     *Get the data type and get() parameter type of the component
+     * @dev SHOULD Import Types Library, which is an enumeration Library containing all data types.
+     * Entity data can be stored according to the data type.
+     * The get() parameter data type can be used to get entity data.
+     * @return the data type array of the entity
+     * @return get parameter data type array
+     */
+    function types()
+        external
+        view
+        returns (Types.Type[] memory, Types.Type[] memory);
+
+    /**
+     *Store entity data.
+     * @dev entity MUST be available. The system that operates on it MUST be available.
+     * The entity has the component attached.
+     * @param _entityId is the Id of the entity.
+     * @param _data is the data to be stored.
+     */
+    function set(uint256 _entityId, bytes memory _data) external;
+
+    /**
+     *Get the data of the entity according to the entity Id.
+     * @param _entityId is the Id of the entity.
+     * @return Entity data.
+     */
+    function get(uint256 _entityId) external view returns (bytes memory);
+
+    /** Get the data of the entity according to the entity Id and parameters.
+     * @param _entityId is the Id of the entity.
+     * @param _params is an extra parameter, it SHOULD depend on whether you need it.
+     * @return Entity data.
+     */
+    function get(uint256 _entityId, bytes memory _params)
+        external
+        view
+        returns (bytes memory);
+}
+```
+
+## Rationale
+
+World contracts are containers for entities, component contracts, and system contracts. Its core principle is to establish the relationship between entities and component contracts, and different entities will attach different components. And use the system contract to dynamically change the data of the entity in the component.
+Usual workflow when building ECS-based programs
+
+1. Implement the `IWorld` interface to create a world contract.
+2. Call `createEntity()` of the world contract to create an entity.
+3. Implement the `IComponent` interface to create a Component contract.
+4. Call `registerComponent()` of the world contract to register the component contract.
+5. Call `addComponent()` of the world contract to attach the component to the entity.
+6. Create a system contract, which is a contract without interface restrictions, and you can define any function in the system contract.
+7. Call `registerSystem()` of the world contract to register the system contract.
+8. Run the system.
```

Comment:
These lines seem more like they explain _how to use the standard_, so they probably belong in the specification section.

The rationale section should explain why specific technical choices were made within the proposal. Some examples of questions you might want to answer in this section are:

- why include type information instead of simple `bytes` arrays?
- why differentiate between a non-existent entity and an entity with false state?
- is it gas efficient to have the `getEntityComponents` function?

These are just examples. Don't feel obligated to explain these particular questions, if they're obvious or irrelevant.